### PR TITLE
Add draft of Characters.unicodeScalars proposal

### PR DESCRIPTION
**Pull request by airspeedswift**

### CODE REVIEW

File: proposals/NNNN-character-unicode-view.md [hunk 1]
```diff
@@ -0,0 +1,90 @@
+# Add `unicodeScalars` property to `Character`
+
+* Proposal: [SE-NNNN](NNNN-character-unicode-view.md)
+* Authors: [Ben Cohen](https://github.com/airspeedswift)
+* Review Manager: TBD
+* Status: **Awaiting review**
+
+## Introduction
+
+This proposal adds a `unicodeScalar` view to `Character`, similar to that on `String`.
```
Comment (benrimmington) [inline proposals/NNNN-character-unicode-view.md]: `unicodeScalar` => `unicodeScalars`

File: proposals/NNNN-character-unicode-view.md [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+
+## Motivation
+
+The `Character` element type of `String` is currently a black box that provides
+little functionality besides comparison, literal construction, and to be used
+as an argument to `String.init`.
+
+Many operations on `String` could be neatly/readbily implemented as operations
```
Comment (benrimmington) [inline proposals/NNNN-character-unicode-view.md]: `readbily` => `readily`

File: proposals/NNNN-character-unicode-view.md [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+But you cannot write this:
+
+```swift
+let ws = CharacterSet.whitespacesAndNewlines
+s.split { $0.unicodeScalars.contains(where: ws.contains) }
+```
+
+## Proposed solution
+
+Add a `unicodeScalars` property to `Character`, presending a lazy view of the
```
Comment (benrimmington) [inline proposals/NNNN-character-unicode-view.md]: `presending` => `presenting`

File: proposals/NNNN-character-unicode-view.md [hunk 4]
```diff
@@ -0,0 +1,90 @@
...
+Many operations on `String` could be neatly/readbily implemented as operations
+on each character in the string, if `Character` exposed its scalars more
+directly. Many useful things can be determined by examining the scalars in a
+grapheme (for example is this an ASCII character?).
+
+For example, today you can write this:
+
+```swift
+let s = "one two three"
+s.split(separator: " ")
```
Comment (benrimmington) [inline proposals/NNNN-character-unicode-view.md]: Swift 3.1 (today):
`s.split(separator: " ")` => `s.characters.split(separator: " ")`

File: proposals/NNNN-character-unicode-view.md [hunk 5]
```diff
@@ -0,0 +1,90 @@
...
+## Proposed solution
+
+Add a `unicodeScalars` property to `Character`, presending a lazy view of the
+scalars in the character, along similar lines to the one on `String`.
+
+Unlike the view on `String`, this will _not_ be a mutable view – it will be
+read-only. The preferred method for creating and manipulating non-literal
+`Character` values will be through `String`. While there may be some good
+use cases to manipulating a `Character` directly, these are outweighed by the 
+complexity of ensuring the invariant that it contain exactly one grapheme.
```
Comment (BasThomas) [inline proposals/NNNN-character-unicode-view.md]: `contain` => `contains`

File: proposals/NNNN-character-unicode-view.md [hunk 6]
```diff
@@ -0,0 +1,90 @@
...
+
+## Detailed design
+
+Add the following nested type to `Character`:
+
+```swift
+extension Character {
+  public struct UnicodeScalarView : BidirectionalCollection {
+    public struct Index
```
Comment (BasThomas) [inline proposals/NNNN-character-unicode-view.md]: I don't think this should be a `struct`?

File: proposals/NNNN-character-unicode-view.md [hunk 7]
```diff
@@ -0,0 +1,90 @@
...
+## Effect on API resilience
+
+Purely additive, so no impact.
+
+## Alternatives considered
+
+Adding other views, such as `utf8` or `utf16`, was considered but not deemed useful
+enough compared to using these operations on `String` instead.
+
+In future, this feature could be used to implement convenience methods such as
```
Comment (BasThomas) [inline proposals/NNNN-character-unicode-view.md]: `In *the* future`

File: proposals/NNNN-character-unicode-view.md (hunk 6 above)
Comment (airspeedswift) [inline proposals/NNNN-character-unicode-view.md]: It probably will be a (nested) struct.

File: proposals/NNNN-character-unicode-view.md (hunk 4 above)
Comment (airspeedswift) [inline proposals/NNNN-character-unicode-view.md]: This was in 4.0 dialect ;)

**Pull request merged**